Sources/TokamakCore/Gestures/Performing/GestureStateGesture.swift

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Szymon on 16/7/2023.
+//
+
+public struct GestureStateGesture<Base: Gesture, State>: Gesture {
+    public typealias Value = Base.Value
+    
+    @GestureState private var gestureState: State
+    private var gesture: Base
+
+    private let updatingBody: (Base.Value, inout State, inout Transaction) -> Void
+    private var onEnded: ((Value) -> Void)?
+    
+    public var body: Base.Body {
+        var gesture = gesture._onChanged(perform: { value in
+            // TODO: Is this transaction working?
```

Comment:
@carson-katri @j-f1 could get help from you here? 
This isn't working as SwiftUI. As a matter of fact, I don't think the animations are working here at all

Reply:
created ticket